views/js/richPassage/xincludeRendererAddStyles.js

```diff
@@ -15,27 +15,40 @@
  *
  * Copyright (c) 2021 (original work) Open Assessment Technologies SA ;
  */
-define(['jquery', 'uri', 'util/url', 'core/dataProvider/request'], function ($, uri, urlUtil, request) {
+define([
+    'lodash',
+    'jquery',
+    'uri',
+    'util/url',
+    'core/dataProvider/request',
+    'taoMediaManager/qtiCreator/helper/formatStyles'
+], function (_, $, uri, urlUtil, request, formatStyles) {
     'use strict';
 
-    return function xincludeRendererAddStyles(passageHref, head = $('head')) {
+    return function xincludeRendererAddStyles(passageHref, passageClassName, serial, head = $('head'), preview = false) {
         if (/taomedia:\/\/mediamanager\//.test(passageHref)) {
             // check rich passage styles and inject them to item
             const passageUri = uri.decode(passageHref.replace('taomedia://mediamanager/', ''));
             request(urlUtil.route('getStylesheets', 'SharedStimulusStyling', 'taoMediaManager'), {
                 uri: passageUri
             })
                 .then(response => {
-                    response.forEach(element => {
+                    response.children.forEach(stylesheet => {
+                        // check different names of stylesheets
+                        const link = urlUtil.route('loadStylesheet', 'SharedStimulusStyling', 'taoMediaManager', {
+                            uri: passageUri,
+                            stylesheet: stylesheet.name
+                        });
                         const styleElem = $('<link>', {
                             rel: 'stylesheet',
                             type: 'text/css',
-                            href: urlUtil.route('loadStylesheet', 'SharedStimulusStyling', 'taoMediaManager', {
-                                uri: passageUri,
-                                stylesheet: element
-                            }),
+                            href: link,
                             'data-serial': passageUri
```

Comment:
```suggestion
                            'data-serial': serial
```

instead of `styleElem[0].dataset['serial'] = serial;`

Reply:
This function is called by the Item Authoring and the Passage Preview on the Shared Stimulus Resource Manager modal on the Item Authoring.
It hasn't got the same scope. If you see, I've done a condition on line 48, and I only need to set the serial and call the `onload` function when the Item authoring calls it. The Resource Manager has an iframe that contains the CSS files, so there is no need to call extra formating functions. And in this case, the serial value is empty.